discovery_child_development/utils/huggingface_pipeline.py

```diff
@@ -148,28 +158,44 @@ def load_training_args(
-        report_to=config["report_to"],
-        learning_rate=config["learning_rate"],
-        per_device_train_batch_size=config["per_device_train_batch_size"],
-        per_device_eval_batch_size=config["per_device_eval_batch_size"],
-        gradient_accumulation_steps=config["gradient_accumulation_steps"],
-        num_train_epochs=config["num_train_epochs"],
-        weight_decay=config["weight_decay"],
-        adam_beta1=config["adam_beta1"],
-        adam_beta2=config["adam_beta2"],
-        adam_epsilon=config["adam_epsilon"],
-        max_grad_norm=config["max_grad_norm"],
-        evaluation_strategy=config["evaluation_strategy"],
-        save_strategy=config["save_strategy"],
-        metric_for_best_model=config["metric_for_best_model"],
-        load_best_model_at_end=config["load_best_model_at_end"],
-        seed=config["seed"],
-        optim=config["optim"],
-    )
+    if config["problem_type"] == "multi_label_classification":
+        return TrainingArguments(
+            output_dir=output_dir,
+            overwrite_output_dir=config["overwrite_output_dir"],
+            report_to=config["report_to"],
+            learning_rate=config["learning_rate"],
+            per_device_train_batch_size=config["per_device_train_batch_size"],
+            per_device_eval_batch_size=config["per_device_eval_batch_size"],
+            gradient_accumulation_steps=config["gradient_accumulation_steps"],
+            num_train_epochs=config["num_train_epochs"],
+            weight_decay=config["weight_decay"],
+            adam_beta1=config["adam_beta1"],
+            adam_beta2=config["adam_beta2"],
+            adam_epsilon=config["adam_epsilon"],
+            max_grad_norm=config["max_grad_norm"],
+            evaluation_strategy=config["evaluation_strategy"],
+            save_strategy=config["save_strategy"],
+            metric_for_best_model=config["metric_for_best_model"],
+            load_best_model_at_end=config["load_best_model_at_end"],
+            seed=config["seed"],
+            optim=config["optim"],
+        )
+    elif config["problem_type"] == "single_label_classification":
+        return TrainingArguments(
+            output_dir=output_dir,
+            report_to=config["report_to"],
+            learning_rate=config["learning_rate"],
+            per_device_train_batch_size=config["per_device_train_batch_size"],
+            per_device_eval_batch_size=config["per_device_eval_batch_size"],
+            gradient_accumulation_steps=config["gradient_accumulation_steps"],
+            num_train_epochs=config["num_train_epochs"],
+            weight_decay=config["weight_decay"],
+            evaluation_strategy=config["evaluation_strategy"],
+            save_strategy=config["save_strategy"],
+            metric_for_best_model=config["metric_for_best_model"],
+            load_best_model_at_end=config["load_best_model_at_end"],
+            seed=config["seed"],
+        )
```

Comment:
Woah, this looks formidable :D Let's keep it as is for now, but I'll keep a note for myself to check if it's possible to simplify this somehow.

Reply:
Yeah its really not ideal, I noticed some changes in dev to the function since I last used it, so as a simple fix just changed it based on which classification problem it was. Thinking about it, the easiest way is probably to have a dictionary of all the possible arguments and then pass that to the TrainingArguments function. I will give that a quick go! Otherwise I'll leave it.